### ISSUE TITLE
How to adjust parameters to realistic pulse profile

### ISSUE THREAD

**Issue by Hai0421** [paulray/NICERsoft#103]
Hi, I want to ask a question, I can't see the period when I fold down (J0030+00451） is a known pair of X-ray pulse sources), which parameter do I need to adjust? There is a detailed tutorial of this software, this software is very useful, thank you very much! 

Below this is my par file and pulse chart

# Created: 2024-10-20T19:10:47.654311
# PINT_version: 1.0.1
# User: hailei
# Host: leihai.local
# OS: macOS-15.0-arm64-arm-64bit
# Python: 3.12.4 | packaged by Anaconda, Inc. | (main, Jun 18 2024, 10:07:17) [Clang 14.0.6 ]
# Format: pint
PSRJ                           J0030+0451
EPHEM                               DE421
CLK                          TT(BIPM2011)
UNITS                                 TDB
START              53331.8428259775636739 0 0.0
FINISH             57976.1329119150636760 0 0.0
TIMEEPH                              FB90
T2CMETHOD                        IAU2000B
DILATEFREQ                              N
DMDATA                                  N
NTOA                                 2845
CHI2R                              1.8068 0 2834.0
TRES                                 3.22
RAJ                      0:30:27.42725720 1 0.00000001115045757510
DECJ                     4:51:39.70757000 1 0.00000038809116409062
PMRA                   -6.288507497153388 1 0.20681023212444358
PMDEC                  0.6155970864728986 1 0.48077763529257733
PX                     2.6583873577563137 1 0.1475626504611243
POSEPOCH           55999.9998006651803735
F0                     205.53069908812228 1 1.5014730632806368e-12
F1                  -4.29804908390079e-16 1 3.2925379580757124e-21
PEPOCH             55999.9998006651803735
TZRMJD             55655.4797077947792824
TZRSITE                                 f
TZRFRQ                        1438.234009
JUMP            TEL nancay     8.036571461288969e-06 1 0.0
PLANET_SHAPIRO                          Y
NE_SW                   4.000000124041585
SWM                                     0
DM                       4.33397101341168 1 0.000338569307285196
DM1                 0.0001505929231760213 1 0.00010677755921359501
DMEPOCH            55999.9998006651803735

![phaseogram](https://github.com/user-attachments/assets/a2cba00d-c1f5-4b62-9f63-b45e84109b7b)

**Comment by Hai0421** [comment]
psrpipe.py --par J0030+0451_tdb.par  --ephem DE405 1060020361

**Comment by Hai0421** [comment]
Is there a tutorial for this software? Hope to share

**Comment by paulray** [comment]
There is a lot of info at this workshop, including a timing tutorial by Megan DeCesar: https://heasarc.gsfc.nasa.gov/docs/ixpe/analysis/workshop/joint2024.html

**Comment by paulray** [comment]
I tried your specific command and I get the same result. There isn't any problem, just that there is quite a bit of background in this observation and you are allowing all energies from 0.25-15.0 keV, so the signal is swamped out. If you restrict your energy cuts to 0.25-1.5 keV you will see the pulsar with an H-test of 96.  You can do this with `--emin 0.25 --emax 1.5` or by using ftselect on the output cleanfilt.evt file.  The way to determine the optimal cuts is to use `nioptcuts.py`. Hope that helps.

**Comment by Hai0421** [comment]
Thank you very much for your response. Additionally, the generated plot shows significant fluctuations in counts (Figure 1). Is it possible to adjust the bin size or download multiple files from the same source to increase exposure, making the profile smoother (like Figure 2)? Such a plot is not suitable for scientific analysis. Thank you for your help
![phaseogram](https://github.com/user-attachments/assets/2acaf523-3af7-49de-a20c-38684c4c59cf)
<img width="706" alt="截屏2024-10-21 22 43 22" src="https://github.com/user-attachments/assets/9eba61a4-155f-41e8-afa4-75c715dcd8c1">

**Comment by paulray** [comment]
Certainly, you are just looking at 1 ObsID. There are several megaseconds of data on this pulsar in the NICER archive.  What are you trying to do? If you just want a high S/N profile from NICER, you could just use the dataset published here: https://zenodo.org/records/3524457

**Comment by Hai0421** [comment]
OK, I seem to understand. Thanks for your patient help

**Comment by Hai0421** [comment]
Can this software directly read the J0030+0451_merged_phase_0.25-3keV.fits file provided in the web page to generate the graph? Or do you need to write python code to read it?

**Comment by paulray** [comment]
That is just a FITS event list that many codes can read, including various FTOOLS. The scripts in NICERsoft like `niphaseogram.py` and `plotphist.py` can read it. It is also very easy to read in python and plot in whatever way you need.

**Comment by Hai0421** [comment]
thank you1
